### ISSUE TITLE
"AWS SAM Hello World (Maven)" fails to deploy with "Unable to upload artifact"

### ISSUE THREAD

**Issue by larsschaps** [aws/aws-toolkit-jetbrains#1407]
**Describe the bug**
Deployment of HelloWorld fails with SSL validation error:

**Masked** messsage:

"C:\Program Files\Amazon\AWSSAMCLI\bin\sam.cmd" package --template-file C:\Users\XXXXX\aldi-tcmapp\test\.aws-sam\build\template.yaml --output-template-file C:\Users\XXXXX\aldi-tcmapp\test\.aws-sam\build\packaged-template.yaml --s3-bucket testtcmapp
Error: Unable to upload artifact HelloWorldFunction referenced by CodeUri parameter of HelloWorldFunction resource.
SSL validation failed for https://aaaaaaaa.s3.amazonaws.com/123456890 [Errno 2] No such file or directory

**To reproduce**

In Intellij
1. Select from the menu "New" -> "Project".
2. Select "AWS".
3. Select "AWS Serverless Application".
4. Press "Next".
5. Enter Project name: "Test123"
6. Set project location to whatever suits you.
7. Runtime "java8"
8. SAM Template: "AWS SAM Hello World (Maven)"
9. Project SDK "JDK1.8 (java version "1.8.0_221"
10. "Finish"
11. Right click "template.yaml"
12. Select "Deploy Serverless Application"
13. Create Stack "Test123"
14. Create Bucket "mytestbucketjhelloworld" or any other valid name
15. Hit "Deploy"

Process stops with above error message.

**Expected behavior**
No error message. Deploymen of helloworld example.

**Your Environment**
 - OS: Windows 10 64bit Version 1809 (Build 17763.7875)
 - JetBrains' Product: Intellij IDEA Ultimate 2019.2.4
 - JetBrains' Product Version: 2019.2.4
 - Toolkit Version: AWS Toolkit 1.9-192
 - SAM CLI Version: SAM CLI, version 0.34.0
 - JVM/Python Version: JDK1.8 (java version "1.8.0_221
 - Apache Maven 3.3.9
 - java version "1.8.0_221"
 - aws-cli/1.16.292 Python/3.6.0 Windows/10 botocore/1.13.28

**Comment by hunterwerlla** [comment]
Are you running behind a proxy?

**Comment by larsschaps** [comment]
Thanks for your comment @hunterwerlla, but I'm afraid not. Actually, you made me made me try in a different network (home, not work - definitely no proxy or firewall here), but same result. :-(

**Comment by hunterwerlla** [comment]
Can you try updating to the SAM cli 0.35? It seems like the version of the SAM CLI on your machine cannot access the trust store

**Comment by rli** [comment]
Do you happen to have a dot in your bucket name? S3 isn't creating valid certificates for those bucket names right now, which might explain the cert validation failure.

**Comment by larsschaps** [comment]
Hi...
@rli , thank for your input. But no, no dot in the s3 bucket.

@hunterwerlla 
When i change the code in botocore/client.py ( ClientCreator::createClient()) so that "verify" is always False, so that no validation is done, i get:

> Successfully packaged artifacts and wrote output template to file C:\Users\SCHAPSL\aldi-tcmapp\AWSSAMHelloWorld\.aws-sam\build\packaged-template.yaml.

Which trust store is actually used? Some of the cacerts.pem of the installation or he Windows trust store?

Thanks,
Lars

**Comment by abrooksv** [comment]
Before SAM 0.31, they used the AWS CLI to handle the communication. 

Can you try to do something like `aws s3 ls`? If that works, it may be a gap in their packaging that the AWS CLI hid.

**Comment by larsschaps** [comment]
Hello everyone..
sorry for the hassle. All my fault.
I found the issue. I was experimenting with "AWS_CA_BUNDLE" to enable a self signed certificate and i thought I removed the environment variable, which i did not. 
However, you helped me finding it. I would have never put a "print(ca_certs, ca_cert_dir)" in 
C:\Program Files\Amazon\AWSCLI\runtime\Lib\site-packages\urllib3\util\ssl_.py ssl_wrap_socket() to find out.
That's the nice thing about script languages ;-)

brgs
Lars
